runtime/src/deterministic.rs

```diff
@@ -544,22 +587,13 @@ pub struct Runner {
 impl crate::Runner for Runner {
     fn start<F>(self, f: F) -> F::Output
     where
-        F: Future + Send + 'static,
-        F::Output: Send + 'static,
+        F: Future,
     {
-        // Add root task to the queue
-        let output = Arc::new(Mutex::new(None));
-        Tasks::register(
-            &self.executor.tasks,
-            "",
-            true,
-            Box::pin({
-                let output = output.clone();
-                async move {
-                    *output.lock().unwrap() = Some(f.await);
-                }
-            }),
-        );
+        // Pin root task to the heap
+        let mut root = Box::pin(f);
```

Comment:
This change created the most complexity.

Now that `f` is neither `Send` nor `'static`, we need to be a lot more careful about where it can be stored.